### ISSUE TITLE
Obfuscate ClassName not change name class on another script under another packagename by Ignore Option

### ISSUE THREAD

**Issue by tracer8** [ClaudiuGeorgiu/Obfuscapk#40]
Hi, i have quick modified Ignore option. Now it will only obfuscate some list on resources\libs_to_ignore.txt instead of ignore it. I think it is better include than ignore, so i only add package i need to obfuscate.

Code:
```
# Get only the smali files that are part of known thirdparty libraries.
if any(
	relative_smali_file.startswith(lib)
	for lib in libs_to_ignore
):
	filtered_smali_files.append(smali_file)
```

### Steps:
1/ My goal is obfuscate all class under this package name:  "com.myclass.ads"

2/ Use Obfuscapk success.

3/ But after success obfuscate and rebuild, my app crash. When i decompile code and check it i saw:
- It is obfuscated all class under my package "com.myclass.ads".  
- "com.myclass.ads" name have been change to "p4d236d9a.pf0f21d36.pa102906f"
- It trigger error when crash: Didn't find class "com.myclass.ads.Models.AdConfigs" on path: DexPathList.

### Reason:
On another script, my "com.myclass.ads" classes is not change. On my MainActivity, under another package, it is still remaining package:
"com.myclass.ads"

So did i miss something? Did i need to modify anything to make it work? To Include than Ignore
Thanks!

**Comment by tracer8** [comment]
When i try to my another packagename too, it is working.
Look like i must add all of my package depending each other to include.txt file to make it is working

**Comment by tracer8** [comment]
OK, i got that problem:
- Obfuscator only obfuscate list of smali files on include list
- App Package Name is union with ClassRename. I have to modify to remove this option. I think better split it to another obfuscator like PackageRename instead of combine with ClassRename

Btw: My issude have been solved! If you have any suggest for me about include option instead of ignore i have modified. Please share your mind. Beside you can close this post.
Many thanks!

**Comment by ClaudiuGeorgiu** [comment]
> * Obfuscator only obfuscate list of smali files on include list

Hi, you got it right. When a class name is changed, the tool will look for its usages in the other smali files and will rename them too. For third party libraries this works well because if you are using a class from a library, when the class in the library is renamed your usage is renamed too. When you ignore a library, there is no reason to even look at its smali file, since it's only third party code that won't be modified by the obfuscation.

However, if you want to just keep some packages and ignore the rest, you have to do a little modification: you have to keep a separate list with all the smali files, and when you rename a class you have to look in all the other smali files and rename its usages too. It's just a more general version of what you did:
> Look like i must add all of my package depending each other to include.txt file to make it is working

To recap: if you exclude class **E** and include the rest, you don't have to look if it is used somewhere, because even if it's used it won't be modified by the obfuscation. If you include class **I** and exclude the rest, you have to look at the other files anyway because maybe it's used somewhere in one of the files that you excluded, and since it will be modified by the obfuscation you have to update all its usages.

**Comment by tracer8** [comment]
Thank you! I got it right!
